Incorrect line terminators in test email files (`\r\r\n`)

Hi!

I'm working on a side project to implement these chatmail helper daemons in Go and wire it up with [the Maddy email server](https://maddy.email) and [GoKrazy](https://gokrazy.org) to create a chatmail appliance. In the process of reimplementing `filtermail.py`, I discovered that my implementation did not pass the PGP literal packet unit test, even though both the implementation and the test were approximately line-for-line copies of the original Python code.

This seems to be caused by the line terminators in the test email files. Instead of `\r\n`, as the RFCs require, they are `\r\r\n`, which fools the Python `email` library into thinking the emails are not multipart. This exits the `check_encrypted()` function immediately, bypassing all of the PGP packet validation code entirely. 

https://github.com/deltachat/chatmail/blob/main/chatmaild/src/chatmaild/filtermail.py#L113-L114

(I had fixed the line endings in my copies of the files while troubleshooting a different error, where the Go equivalent email library rejected the input files as invalid because there was no `\r\n\r\n` separator between the email headers and the email body.)

When I fixed the line endings in all of the affected test email files in this repository and then re-ran the tests, this is the output I received:

```
lint: commands[0]> ruff format --quiet --diff src/
lint: commands[1]> ruff check src/
All checks passed!
lint: OK ✔ in 0.13 seconds
.pkg: _optional_hooks> python '/Users/s0ph0s/Library/Application Support/pipx/.cache/29bc9cf211a290c/lib/python3.12/site-packages/pyproject_api/_backend.py' True setuptools.build_meta
.pkg: get_requires_for_build_sdist> python '/Users/s0ph0s/Library/Application Support/pipx/.cache/29bc9cf211a290c/lib/python3.12/site-packages/pyproject_api/_backend.py' True setuptools.build_meta
.pkg: build_sdist> python '/Users/s0ph0s/Library/Application Support/pipx/.cache/29bc9cf211a290c/lib/python3.12/site-packages/pyproject_api/_backend.py' True setuptools.build_meta
py: install_package> python -I -m pip install --force-reinstall --no-deps /Users/s0ph0s/Developer/Python/chatmail/chatmaild/.tox/.tmp/package/11/chatmaild-0.2.tar.gz
py: commands[0]> pytest -v -rsXx
============================= test session starts ==============================
platform darwin -- Python 3.12.5, pytest-8.3.3, pluggy-1.5.0 -- /Users/s0ph0s/Developer/Python/chatmail/chatmaild/.tox/py/bin/python
cachedir: .tox/py/.pytest_cache
rootdir: /Users/s0ph0s/Developer/Python/chatmail/chatmaild
configfile: pyproject.toml
plugins: deltachat-rpc-client-1.147.1, chatmaild-0.2
collecting ... collected 57 items

src/chatmaild/tests/test_config.py::test_read_config_basic PASSED        [  1%]
src/chatmaild/tests/test_config.py::test_read_config_testrun PASSED      [  3%]
src/chatmaild/tests/test_config.py::test_config_userstate_paths PASSED   [  5%]
src/chatmaild/tests/test_config.py::test_config_max_message_size PASSED  [  7%]
src/chatmaild/tests/test_delete_inactive_users.py::test_login_timestamps PASSED [  8%]
src/chatmaild/tests/test_delete_inactive_users.py::test_delete_inactive_users PASSED [ 10%]
src/chatmaild/tests/test_doveauth.py::test_basic PASSED                  [ 12%]
src/chatmaild/tests/test_doveauth.py::test_iterate_addresses PASSED      [ 14%]
src/chatmaild/tests/test_doveauth.py::test_invalid_username_length PASSED [ 15%]
src/chatmaild/tests/test_doveauth.py::test_dont_overwrite_password_on_wrong_login PASSED [ 17%]
src/chatmaild/tests/test_doveauth.py::test_nocreate_file PASSED          [ 19%]
src/chatmaild/tests/test_doveauth.py::test_handle_dovecot_request PASSED [ 21%]
src/chatmaild/tests/test_doveauth.py::test_handle_dovecot_protocol_hello_is_skipped PASSED [ 22%]
src/chatmaild/tests/test_doveauth.py::test_handle_dovecot_protocol_user_not_exists PASSED [ 24%]
src/chatmaild/tests/test_doveauth.py::test_handle_dovecot_protocol_iterate PASSED [ 26%]
src/chatmaild/tests/test_doveauth.py::test_50_concurrent_lookups_different_accounts PASSED [ 28%]
src/chatmaild/tests/test_filedict.py::test_basic PASSED                  [ 29%]
src/chatmaild/tests/test_filedict.py::test_bad_marshal_file PASSED       [ 31%]
src/chatmaild/tests/test_filedict.py::test_write_bytes_atomic_concurrent PASSED [ 33%]
src/chatmaild/tests/test_filtermail.py::test_reject_forged_from PASSED   [ 35%]
src/chatmaild/tests/test_filtermail.py::test_filtermail_no_encryption_detection PASSED [ 36%]
src/chatmaild/tests/test_filtermail.py::test_filtermail_encryption_detection PASSED [ 38%]
src/chatmaild/tests/test_filtermail.py::test_filtermail_no_literal_packets FAILED [ 40%]
src/chatmaild/tests/test_filtermail.py::test_filtermail_unencrypted_mdn PASSED [ 42%]
src/chatmaild/tests/test_filtermail.py::test_send_rate_limiter PASSED    [ 43%]
src/chatmaild/tests/test_filtermail.py::test_excempt_privacy PASSED      [ 45%]
src/chatmaild/tests/test_filtermail.py::test_passthrough_senders PASSED  [ 47%]
src/chatmaild/tests/test_filtermail.py::test_check_armored_payload PASSED [ 49%]
src/chatmaild/tests/test_lastlogin.py::test_handle_dovecot_request_last_login PASSED [ 50%]
src/chatmaild/tests/test_lastlogin.py::test_handle_dovecot_request_last_login_echobot PASSED [ 52%]
src/chatmaild/tests/test_metadata.py::test_metadata_persistence PASSED   [ 54%]
src/chatmaild/tests/test_metadata.py::test_remove_nonexisting PASSED     [ 56%]
src/chatmaild/tests/test_metadata.py::test_notifier_remove_without_set PASSED [ 57%]
src/chatmaild/tests/test_metadata.py::test_handle_dovecot_request_lookup_fails PASSED [ 59%]
src/chatmaild/tests/test_metadata.py::test_handle_dovecot_request_happy_path PASSED [ 61%]
src/chatmaild/tests/test_metadata.py::test_handle_dovecot_protocol_set_devicetoken PASSED [ 63%]
src/chatmaild/tests/test_metadata.py::test_handle_dovecot_protocol_set_get_devicetoken PASSED [ 64%]
src/chatmaild/tests/test_metadata.py::test_handle_dovecot_protocol_iterate PASSED [ 66%]
src/chatmaild/tests/test_metadata.py::test_notifier_thread_deletes_persistent_file PASSED [ 68%]
src/chatmaild/tests/test_metadata.py::test_notifier_thread_connection_failures[status0] PASSED [ 70%]
src/chatmaild/tests/test_metadata.py::test_notifier_thread_connection_failures[404] PASSED [ 71%]
src/chatmaild/tests/test_metadata.py::test_notifier_thread_connection_failures[500] PASSED [ 73%]
src/chatmaild/tests/test_metadata.py::test_requeue_removes_tmp_files PASSED [ 75%]
src/chatmaild/tests/test_metadata.py::test_start_and_stop_notification_threads PASSED [ 77%]
src/chatmaild/tests/test_metadata.py::test_multi_device_notifier PASSED  [ 78%]
src/chatmaild/tests/test_metadata.py::test_notifier_thread_run_gone_removes_token PASSED [ 80%]
src/chatmaild/tests/test_metadata.py::test_persistent_queue_items PASSED [ 82%]
src/chatmaild/tests/test_metadata.py::test_iroh_relay PASSED             [ 84%]
src/chatmaild/tests/test_metrics.py::test_main PASSED                    [ 85%]
src/chatmaild/tests/test_migrate_db.py::test_migration_not_exists PASSED [ 87%]
src/chatmaild/tests/test_migrate_db.py::test_migration PASSED            [ 89%]
src/chatmaild/tests/test_newmail.py::test_create_newemail_dict PASSED    [ 91%]
src/chatmaild/tests/test_newmail.py::test_print_new_account PASSED       [ 92%]
src/chatmaild/tests/test_user.py::test_login_timestamp PASSED            [ 94%]
src/chatmaild/tests/test_user.py::test_get_user_dict_not_set PASSED      [ 96%]
src/chatmaild/tests/test_user.py::test_get_user_dict PASSED              [ 98%]
src/chatmaild/tests/test_user.py::test_no_mailboxes_dir PASSED           [100%]

=================================== FAILURES ===================================
______________________ test_filtermail_no_literal_packets ______________________

maildata = <function maildata.<locals>.maildata at 0x1076aac00>

    def test_filtermail_no_literal_packets(maildata):
        """Test that literal OpenPGP packet is not considered an encrypted mail."""
        msg = maildata("literal.eml", from_addr="1@example.org", to_addr="2@example.org")
        # breakpoint()
>       assert not check_encrypted(msg)
E       assert not True
E        +  where True = check_encrypted(<email.message.EmailMessage object at 0x106b22870>)

src/chatmaild/tests/test_filtermail.py:77: AssertionError
=============================== warnings summary ===============================
.tox/py/lib/python3.12/site-packages/chatmaild/doveauth.py:1
  /Users/s0ph0s/Developer/Python/chatmail/chatmaild/.tox/py/lib/python3.12/site-packages/chatmaild/doveauth.py:1: DeprecationWarning: 'crypt' is deprecated and slated for removal in Python 3.13
    import crypt

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=================== 1 failed, 56 passed, 1 warning in 2.45s ====================
py: exit 1 (3.31 seconds) /Users/s0ph0s/Developer/Python/chatmail/chatmaild> pytest -v -rsXx pid=13843
  lint: OK (0.13=setup[0.04]+cmd[0.04,0.04] seconds)
  py: FAIL code 1 (8.75=setup[5.45]+cmd[3.31] seconds)
  evaluation failed :( (8.95 seconds)
```

Then I checked `git blame` and discovered that [the test files have been unchanged since they were originally committed](https://github.com/deltachat/chatmail/blame/main/chatmaild/src/chatmaild/tests/mail-data/literal.eml), which means that this test never worked.

In order to fix the underlying bug, I think the right solution is to replace this part of the PGP validation code:
https://github.com/deltachat/chatmail/blob/main/chatmaild/src/chatmaild/filtermail.py#L63-L66

with `return packet_type_id == 18`, but I'm not particularly experienced with PGP and I'd like you all to double-check me before I open a PR that fixes the line endings and makes that replacement!

Thank you for all the hard work you're putting into this project—it's very cool, and I'm hopeful that it will catch on :)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -39,6 +39,9 @@
 - add IMAP capabilities instead of overwriting them
   ([#413](https://github.com/deltachat/chatmail/pull/413))
 
+- fix OpenPGP payload check
+  ([#435](https://github.com/deltachat/chatmail/pull/435))
+
 
 ## 1.4.1 2024-07-31
 

--- a/chatmaild/src/chatmaild/filtermail.py
+++ b/chatmaild/src/chatmaild/filtermail.py
@@ -60,10 +60,11 @@ def check_openpgp_payload(payload: bytes):
         i += body_len
 
         if i == len(payload):
-            if packet_type_id == 18:
-                # Last packet should be
-                # Symmetrically Encrypted and Integrity Protected Data Packet (SEIPD)
-                return True
+            # Last packet should be
+            # Symmetrically Encrypted and Integrity Protected Data Packet (SEIPD)
+            #
+            # This is the only place where this function may return `True`.
+            return packet_type_id == 18
         elif packet_type_id not in [1, 3]:
             # All packets except the last one must be either
             # Public-Key Encrypted Session Key Packet (PKESK)
@@ -71,13 +72,7 @@ def check_openpgp_payload(payload: bytes):
             # Symmetric-Key Encrypted Session Key Packet (SKESK)
             return False
 
-    if i == 0:
-        return False
-
-    if i > len(payload):
-        # Payload is truncated.
-        return False
-    return True
+    return False
 
 
 def check_armored_payload(payload: str):

--- a/chatmaild/src/chatmaild/tests/mail-data/literal.eml
+++ b/chatmaild/src/chatmaild/tests/mail-data/literal.eml
@@ -1,44 +1,44 @@
-From: {from_addr}
-To: {to_addr}
-Subject: ...
-Date: Sun, 15 Oct 2023 16:43:21 +0000
-Message-ID: <Mr.UVyJWZmkCKM.hGzNc6glBE_@c2.testrun.org>
-In-Reply-To: <Mr.MvmCz-GQbi_.6FGRkhDf05c@c2.testrun.org>
-References: <Mr.3gckbNy5bch.uK3Hd2Ws6-w@c2.testrun.org>
-	<Mr.MvmCz-GQbi_.6FGRkhDf05c@c2.testrun.org>
-Chat-Version: 1.0
-Autocrypt: addr={from_addr}; prefer-encrypt=mutual;
-	keydata=xjMEZSwWjhYJKwYBBAHaRw8BAQdAQBEhqeJh0GueHB6kF/DUQqYCxARNBVokg/AzT+7LqH
-	rNFzxiYXJiYXpAYzIudGVzdHJ1bi5vcmc+wosEEBYIADMCGQEFAmUsFo4CGwMECwkIBwYVCAkKCwID
-	FgIBFiEEFTfUNvVnY3b9F7yHnmme1PfUhX8ACgkQnmme1PfUhX9A4AEAnHWHp49eBCMHK5t66gYPiW
-	XQuB1mwUjzGfYWB+0RXUoA/0xcQ3FbUNlGKW7Blp6eMFfViv6Mv2d3kNSXACB6nmcMzjgEZSwWjhIK
-	KwYBBAGXVQEFAQEHQBpY5L2M1XHo0uxf8SX1wNLBp/OVvidoWHQF2Jz+kJsUAwEIB8J4BBgWCAAgBQ
-	JlLBaOAhsMFiEEFTfUNvVnY3b9F7yHnmme1PfUhX8ACgkQnmme1PfUhX/INgEA37AJaNvruYsJVanP
-	IXnYw4CKd55UAwl8Zcy+M2diAbkA/0fHHcGV4r78hpbbL1Os52DPOdqYQRauIeJUeG+G6bQO
-MIME-Version: 1.0
-Content-Type: multipart/encrypted; protocol="application/pgp-encrypted";
-	boundary="YFrteb74qSXmggbOxZL9dRnhymywAi"
-
-
---YFrteb74qSXmggbOxZL9dRnhymywAi
-Content-Description: PGP/MIME version identification
-Content-Type: application/pgp-encrypted
-
-Version: 1
-
-
---YFrteb74qSXmggbOxZL9dRnhymywAi
-Content-Description: OpenPGP encrypted message
-Content-Disposition: inline; filename="encrypted.asc";
-Content-Type: application/octet-stream; name="encrypted.asc"
-
------BEGIN PGP MESSAGE-----
-
-yxJiAAAAAABIZWxsbyB3b3JsZCE=
-=1I/B
------END PGP MESSAGE-----
-
-
---YFrteb74qSXmggbOxZL9dRnhymywAi--
-
+From: {from_addr}
+To: {to_addr}
+Subject: ...
+Date: Sun, 15 Oct 2023 16:43:21 +0000
+Message-ID: <Mr.UVyJWZmkCKM.hGzNc6glBE_@c2.testrun.org>
+In-Reply-To: <Mr.MvmCz-GQbi_.6FGRkhDf05c@c2.testrun.org>
+References: <Mr.3gckbNy5bch.uK3Hd2Ws6-w@c2.testrun.org>
+	<Mr.MvmCz-GQbi_.6FGRkhDf05c@c2.testrun.org>
+Chat-Version: 1.0
+Autocrypt: addr={from_addr}; prefer-encrypt=mutual;
+	keydata=xjMEZSwWjhYJKwYBBAHaRw8BAQdAQBEhqeJh0GueHB6kF/DUQqYCxARNBVokg/AzT+7LqH
+	rNFzxiYXJiYXpAYzIudGVzdHJ1bi5vcmc+wosEEBYIADMCGQEFAmUsFo4CGwMECwkIBwYVCAkKCwID
+	FgIBFiEEFTfUNvVnY3b9F7yHnmme1PfUhX8ACgkQnmme1PfUhX9A4AEAnHWHp49eBCMHK5t66gYPiW
+	XQuB1mwUjzGfYWB+0RXUoA/0xcQ3FbUNlGKW7Blp6eMFfViv6Mv2d3kNSXACB6nmcMzjgEZSwWjhIK
+	KwYBBAGXVQEFAQEHQBpY5L2M1XHo0uxf8SX1wNLBp/OVvidoWHQF2Jz+kJsUAwEIB8J4BBgWCAAgBQ
+	JlLBaOAhsMFiEEFTfUNvVnY3b9F7yHnmme1PfUhX8ACgkQnmme1PfUhX/INgEA37AJaNvruYsJVanP
+	IXnYw4CKd55UAwl8Zcy+M2diAbkA/0fHHcGV4r78hpbbL1Os52DPOdqYQRauIeJUeG+G6bQO
+MIME-Version: 1.0
+Content-Type: multipart/encrypted; protocol="application/pgp-encrypted";
+	boundary="YFrteb74qSXmggbOxZL9dRnhymywAi"
+
+
+--YFrteb74qSXmggbOxZL9dRnhymywAi
+Content-Description: PGP/MIME version identification
+Content-Type: application/pgp-encrypted
+
+Version: 1
+
+
+--YFrteb74qSXmggbOxZL9dRnhymywAi
+Content-Description: OpenPGP encrypted message
+Content-Disposition: inline; filename="encrypted.asc";
+Content-Type: application/octet-stream; name="encrypted.asc"
+
+-----BEGIN PGP MESSAGE-----
+
+yxJiAAAAAABIZWxsbyB3b3JsZCE=
+=1I/B
+-----END PGP MESSAGE-----
+
+
+--YFrteb74qSXmggbOxZL9dRnhymywAi--
+
 

